src/session/session_dhandle.c

```diff
@@ -168,6 +166,12 @@ __wt_session_lock_dhandle(WT_SESSION_IMPL *session, uint32_t flags, bool *is_dea
                 return (0);
             }
 
+            /* Check the special flags again in case they have changed since the last check. */
```

Comment:
This is surprising enough that I think it deserves a more detailed description. Something along the lines of "If we raced with another thread opening the handle, it might not be in the state we need. Check the special flags again to detect that case."

Reply:
We realized we misunderstood part of the problem as so will be removing this check.